pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go

```diff
@@ -4304,3 +4311,9 @@ func withBlockMultiQueue(b *bool) libvmi.Option {
 		vmi.Spec.Domain.Devices.BlockMultiQueue = b
 	}
 }
+
+func withGPU(gpu v1.GPU) libvmi.Option {
```

Comment:
Can you explain why? I'd like to avoid overriding the field.

Reply:
Which other component sets this field?

Reply:
> but I'd like to maintain the design libvmi is going for.

Could you please elaborate?

Reply:
> Could you please elaborate?

All existing libvmi functions that alter slice fields append to the slice rather than override it. The design itself is debatable (perhaps I'm overlooking something) as libvmi is used for construction for the most part (this is especially noticeable when trying to construct a VMI with multiple labels/annotations). While it isn't a written clause, my implementation conforms to that.

Having said all that, I could care less for semantics in this specific case.

Reply:
Thank you for the explanation.
I won't block on this one.